compiler/typecheck/TcEnv.hs

```diff
@@ -548,36 +550,34 @@ tcExtendIdEnv :: [TcId] -> TcM a -> TcM a
 tcExtendIdEnv ids thing_inside
   = tcExtendIdEnv2 [(idName id, id) | id <- ids] thing_inside
 
-tcExtendIdEnv1 :: Name -> Scaled TcId -> TcM a -> TcM a
--- Like tcExtendIdEnv2, but for a single (name,id) pair
--- and checks scaling
-tcExtendIdEnv1 name (Scaled id_mult id) thing_inside
-  = do { (local_usage, result) <- tcCollectingUsage $ tcExtendIdEnv2 [(name,id)] thing_inside
-       ; check_then_add_usage local_usage
-       ; return result }
+tcExtendIdEnv1 :: Name -> TcId -> TcM a -> TcM a
+-- Exactly like tcExtendIdEnv2, but for a single (name,id) pair
+tcExtendIdEnv1 name id thing_inside
+  = tcExtendIdEnv2 [(name,id)] thing_inside
+
+tcExtendIdEnv1Scaled :: Name -> Scaled TcId -> TcM a -> TcM (a, HsWrapper)
```

Comment:
Simon suggested, instead, to have a `checkUsage` function or such, and simply call `checkUsage . tcExtendIdEnv1` where relevant.